File: ltexls-core/src/main/java/org/bsplines/ltexls/tools/ExcludeFromGeneratedCoverage.java
/* Copyright (C) 2020 Julian Valentin, LTeX Development Community
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package org.bsplines.ltexls.tools;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

// Currently, the only way to exclude code from coverage is having an annotation that
// contains the word "Generated" (see https://github.com/jacoco/jacoco/pull/822).
@Retention(RetentionPolicy.CLASS)
@Target({ElementType.TYPE, ElementType.METHOD})
public @interface ExcludeFromGeneratedCoverage {
}

File: src/main/kotlin/org/bsplines/ltexls/tools/ExcludeFromGeneratedCoverage.kt
/* Copyright (C) 2020 Julian Valentin, LTeX Development Community
 *
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 */

package org.bsplines.ltexls.tools

// Currently, the only way to exclude code from coverage is having an annotation that
// contains the word "Generated" (see https://github.com/jacoco/jacoco/pull/822).
@Retention(AnnotationRetention.BINARY)
@Target(AnnotationTarget.TYPE, AnnotationTarget.FUNCTION)
annotation class ExcludeFromGeneratedCoverage
